src/test/java/com/exactpro/th2/sailfish/utils/ImplementationIFilterTest.java

```diff
@@ -0,0 +1,191 @@
+                .build();
+
+        ComparisonResult result = getResult(actual, filter);
+
+        Assertions.assertEquals(status, result.getResult("compareFilter").getStatus());
+    }
+
+    private static List<Arguments> filterError() {
+        return Arrays.asList(
+                Arguments.of("10.1", "10,2", StatusType.FAILED, FilterOperation.LESS, "Failed to parse value to Number. Value = 10,2"),
+                Arguments.of("2007-12-03T10:15:30", "2007-12-03T10-15-30", StatusType.FAILED, FilterOperation.MORE, "Failed to parse value to Date. Value = 2007-12-03T10-15-30"),
+                Arguments.of("2007-12-03", "2007-12-03T10:15:30", StatusType.FAILED, FilterOperation.NOT_MORE, "Failed to compare Temporal values {2007-12-03T10:15:30}, {2007-12-03}")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("filterError")
+    void testFilterWithError(String first, String second, StatusType status, FilterOperation operation, String errorMessage) {
+        RootMessageFilter filter = RootMessageFilter.newBuilder()
+                .setMessageType(MESSAGE_TYPE)
+                .setMessageFilter(MessageFilter.newBuilder()
+                        .putFields("errorFilter", ValueFilter.newBuilder()
+                                .setSimpleFilter(first)
+                                .setOperation(operation)
+                                .build())
+                        .build())
+                .build();
+
+        Message actual = createMessageBuilder(MESSAGE_TYPE)
+                .putFields("errorFilter", getSimpleValue(second))
+                .build();
+
+        ComparisonResult result = getResult(actual, filter);
+
+        Assertions.assertEquals(status, result.getResult("errorFilter").getStatus());
+        Assertions.assertEquals(errorMessage, result.getResult("errorFilter").getExceptionMessage());
+    }
+
+    private static List<Arguments> filterException() {
+        return Arrays.asList(
+                Arguments.of("10,1", "102"),
+                Arguments.of("2007-12-03T10-15:30", "2007-12-03T10:15:30")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("filterException")
+    void testFilterWithException(String first, String second) {
+        RootMessageFilter filter = RootMessageFilter.newBuilder()
+                .setMessageType(MESSAGE_TYPE)
+                .setMessageFilter(MessageFilter.newBuilder()
+                        .putFields("filterException", ValueFilter.newBuilder()
+                                .setSimpleFilter(first)
+                                .setOperation(FilterOperation.MORE)
+                                .build())
+                        .build())
+                .build();
+
+        Message actual = createMessageBuilder(MESSAGE_TYPE)
+                .putFields("filterException", getSimpleValue(second))
+                .build();
+
+        Assertions.assertThrows(IllegalArgumentException.class, () -> getResult(actual, filter));
```

Comment:
I think we don't need the actual message here at all. The exception will be thrown on `converter.fromProtoFilter` so we can use only that method in the assertion. Because of that we method will have only one parameter and you will be able to use `ValueSource` annotation instead of `MethodSource`